src/providers/distributors.tsx

```diff
@@ -0,0 +1,98 @@
+import { useApolloClient } from '@apollo/client'
+import React, { useCallback, useContext, useEffect, useRef, useState } from 'react'
+
+import {
+  GetDistributionBucketsWithOperatorsDocument,
+  GetDistributionBucketsWithOperatorsQuery,
+  GetDistributionBucketsWithOperatorsQueryVariables,
+} from '@/api/queries/__generated__/storage.generated'
+import { ViewErrorFallback } from '@/components/ViewErrorFallback'
+import { DistributorInfo } from '@/types/storage'
+import { ConsoleLogger, SentryLogger } from '@/utils/logs'
+
+type DistributorsMapping = Record<string, DistributorInfo[]>
+
+type DistributorsContextValue = {
+  distributorsMappingPromiseRef: React.MutableRefObject<Promise<DistributorsMapping> | undefined>
+}
+const DistributorsContext = React.createContext<DistributorsContextValue | undefined>(undefined)
+DistributorsContext.displayName = 'DistributorsContext'
+
+export const DistributorsContextProvider: React.FC = ({ children }) => {
+  const distributorsMappingPromiseRef = useRef<Promise<DistributorsMapping>>()
+  const [distributorsError, setDistributorsError] = useState<unknown>(null)
+
+  const client = useApolloClient()
+
+  useEffect(() => {
```

Comment:
Im guessing this effect fetches storage providers but a short comment description above the effect wouldn't hurt

Reply:
added a comment